File: solutions/haiku-1/haiku/java/rle.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        StringBuilder result = new StringBuilder();

        if (input.length() == 0) {
            System.out.println(result.toString());
            return;
        }

        char currentChar = input.charAt(0);
        int count = 1;

        for (int i = 1; i < input.length(); i++) {
            if (input.charAt(i) == currentChar) {
                count++;
            } else {
                result.append(currentChar).append(count);
                currentChar = input.charAt(i);
                count = 1;
            }
        }

        // Append the last run
        result.append(currentChar).append(count);

        System.out.println(result.toString());
    }
}

File: solutions/haiku-1/haiku/perl/rle.pl
#!/usr/bin/perl
use strict;
use warnings;

my $line = <STDIN>;
chomp($line) if defined $line;
$line //= '';  # Default to empty string if undef

my $result = '';
my $current_char = '';
my $count = 0;

for my $char (split //, $line) {
    if ($char eq $current_char) {
        $count++;
    } else {
        if ($current_char ne '') {
            $result .= $current_char . $count;
        }
        $current_char = $char;
        $count = 1;
    }
}

# Don't forget the last run
if ($current_char ne '') {
    $result .= $current_char . $count;
}

print $result;
